aws-distro-opentelemetry-node-autoinstrumentation/test/sampler/aws-xray-remote-sampler.test.ts

```diff
@@ -187,16 +186,66 @@ describe('AwsXrayRemoteSampler', () => {
         }
         expect(sampled).toEqual(100);
         // reset function
-        (AwsXRayRemoteSampler.prototype as any).getDefaultTargetPollingInterval = tmp;
+        (_AwsXRayRemoteSampler.prototype as any).getDefaultTargetPollingInterval = tmp;
         clock.restore();
         done();
       }, 2000);
     }, 100);
   });
 
   it('generates valid ClientId', () => {
-    const clientId: string = (AwsXRayRemoteSampler as any).generateClientId();
+    const clientId: string = (_AwsXRayRemoteSampler as any).generateClientId();
     const match: RegExpMatchArray | null = clientId.match(/[0-9a-z]{24}/g);
     expect(match).not.toBeNull();
   });
+
+  it('toString()', () => {
+    expect(new AwsXRayRemoteSampler({ resource: Resource.EMPTY }).toString()).toEqual(
+      'AwsXRayRemoteSampler{root=ParentBased{root=_AwsXRayRemoteSampler{awsProxyEndpoint=http://localhost:2000, rulePollingIntervalMillis=300000}, remoteParentSampled=AlwaysOnSampler, remoteParentNotSampled=AlwaysOffSampler, localParentSampled=AlwaysOnSampler, localParentNotSampled=AlwaysOffSampler}'
+    );
+  });
+
+  it('ParentBased AwsXRayRemoteSampler creates expected Statistics from the 1 Span with no Parent, disregarding 2 Child Spans', done => {
+    const defaultRuleDir = __dirname + '/data/get-sampling-rules-response-sample-sample-all.json';
+    nock(TEST_URL).post('/GetSamplingRules').reply(200, require(defaultRuleDir));
+
+    const sampler: AwsXRayRemoteSampler = new AwsXRayRemoteSampler({ resource: Resource.EMPTY });
+    const tracerProvider: NodeTracerProvider = new NodeTracerProvider({ sampler: sampler });
+    const tracer: Tracer = tracerProvider.getTracer('test');
+
+    setTimeout(() => {
+      tracer.startActiveSpan('test0', span0 => {
+        const span1: Span = tracer.startSpan('test1');
+        const span2: Span = tracer.startSpan('test2');
+        span2.end();
+        span1.end();
+        span0.end();
+        expect((sampler as any)._root._root.ruleCache.ruleAppliers[0].statistics.RequestCount).toBe(1);
+        expect((sampler as any)._root._root.ruleCache.ruleAppliers[0].statistics.SampleCount).toBe(1);
+        done();
+      });
+    }, 100);
+  });
+
+  it('Non-ParentBased _AwsXRayRemoteSampler creates expected Statistics based on all 3 Spans, disregarding Parent Span Sampling Decision', done => {
+    const defaultRuleDir = __dirname + '/data/get-sampling-rules-response-sample-sample-all.json';
+    nock(TEST_URL).post('/GetSamplingRules').reply(200, require(defaultRuleDir));
+
+    const sampler: _AwsXRayRemoteSampler = new _AwsXRayRemoteSampler({ resource: Resource.EMPTY });
+    const tracerProvider: NodeTracerProvider = new NodeTracerProvider({ sampler: sampler });
+    const tracer: Tracer = tracerProvider.getTracer('test');
+
+    setTimeout(() => {
+      tracer.startActiveSpan('test0', span0 => {
+        const span1: Span = tracer.startSpan('test1');
+        const span2: Span = tracer.startSpan('test2');
+        span2.end();
```

Comment:
Could you explain why span 1 and 2 are not children of span 0? Maybe add some comments here?

Reply:
In both tests, span 1 and 2 are both children of span 0. Will add comments to clarify.

Reply:
I now understand the difference between the two test cases. The comments above _AwsXRayRemoteSampler helps!